client/jetpack-onboarding/steps/stats.jsx

```diff
@@ -0,0 +1,132 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import FormattedHeader from 'components/formatted-header';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import Tile from 'components/tile-grid/tile';
+import TileGrid from 'components/tile-grid';
+import { addQueryArgs } from 'lib/route';
+import { getUnconnectedSiteUrl } from 'state/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { JETPACK_ONBOARDING_STEPS as STEPS } from '../constants';
+import { saveJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
+
+class JetpackOnboardingStatsStep extends React.Component {
+	componentDidUpdate() {
+		const { action, activatedStats, isConnected } = this.props;
+
+		if ( isConnected && activatedStats === false && action === 'activate_stats' ) {
```

Comment:
As soon as JPC actually redirects back to this step, we should also double-check to ensure `this.activateStats()` isn't actually called more than once. I think there's a slight chance that some unrelated prop might change and trigger this lifecycle method while `saveJetpackOnboardingSettings()` is still in-flight, i.e. before `activatedStats` is updated to `true` in `connect()`.

Reply:
I've added some additional checks to avoid such cases in https://github.com/Automattic/wp-calypso/pull/22621/commits/212ddfae1147794320a8613b1756402702ee45f3